src/entities/Snapshot/utils.ts

```diff
@@ -45,3 +45,39 @@ export function calculateMatch(votes1: SnapshotVote[] | null, votes2: SnapshotVo
   }
   return match
 }
+
+export function median(array: number[]) {
+  if (array.length === 0) throw new Error('Median: no inputs')
+
+  array.sort((a, b) => a - b)
+  const half = Math.floor(array.length / 2)
+
+  if (array.length % 2) {
+    return array[half]
+  }
+
+  return (array[half - 1] + array[half]) / 2.0
+}
+
+export function groupProposalsByMonth(proposals: Partial<SnapshotProposal>[], field: keyof SnapshotProposal) {
+  const data: Record<string, number[]> = {}
+
+  for (const proposal of proposals) {
```

Comment:
I like the creation date validation.
I'm thinking it might be event better to catch the failing proposal inside the `for` loop and don't throw an error but rather log and continue processing the others (if that makes sense):

```typescript
for (const proposal of proposals) {
  if (proposal.created) {
       const proposalDate = new Date(proposal.created * 1000)
       const month = proposalDate.getMonth()
       const year = proposalDate.getFullYear()
       const key = `${year}/${month + 1}`
       const value = proposal[field]
       if (typeof value === 'number') {
          data[key] = [...(data[key] || []), value]
       } else {
         console.log(`Proposal field is not a number: ${value}`)  // TODO: rollbar/newrelic
       }
    } else {
      console.log('Proposal has no creation date') // TODO: rollbar/newrelic
    }
 }
```